ignite/chainconfig/config/config.go

```diff
@@ -0,0 +1,175 @@
+
+	// ThirdPartyPath is the relative path of where the third party proto files are
+	// located that used by the app.
+	ThirdPartyPaths []string `yaml:"third_party_paths"`
+}
+
+// Client configures code generation for clients.
+type Client struct {
+	// Vuex configures code generation for Vuex.
+	Vuex Vuex `yaml:"vuex,omitempty"`
+
+	// Dart configures client code generation for Dart.
+	Dart Dart `yaml:"dart,omitempty"`
+
+	// OpenAPI configures OpenAPI spec generation for API.
+	OpenAPI OpenAPI `yaml:"openapi,omitempty"`
+}
+
+// Vuex configures code generation for Vuex.
+type Vuex struct {
+	// Path configures out location for generated Vuex code.
+	Path string `yaml:"path"`
+}
+
+// Dart configures client code generation for Dart.
+type Dart struct {
+	// Path configures out location for generated Dart code.
+	Path string `yaml:"path"`
+}
+
+// OpenAPI configures OpenAPI spec generation for API.
+type OpenAPI struct {
+	Path string `yaml:"path"`
+}
+
+// Faucet configuration.
+type Faucet struct {
+	// Name is faucet account's name.
+	Name *string `yaml:"name"`
+
+	// Coins holds type of coin denoms and amounts to distribute.
+	Coins []string `yaml:"coins"`
+
+	// CoinsMax holds of chain denoms and their max amounts that can be transferred to single user.
+	CoinsMax []string `yaml:"coins_max,omitempty"`
+
+	// LimitRefreshTime sets the timeframe at the end of which the limit will be refreshed
+	RateLimitWindow string `yaml:"rate_limit_window,omitempty"`
+
+	// Host is the host of the faucet server
+	Host string `yaml:"host,omitempty"`
+
+	// Port number for faucet server to listen at.
+	Port int `yaml:"port,omitempty"`
+}
+
+// Init overwrites sdk configurations with given values.
+type Init struct {
+	// App overwrites appd's config/app.toml configs.
+	App map[string]interface{} `yaml:"app"`
+
+	// Client overwrites appd's config/client.toml configs.
+	Client map[string]interface{} `yaml:"client"`
+
+	// Config overwrites appd's config/config.toml configs.
+	Config map[string]interface{} `yaml:"config"`
+
+	// Home overwrites default home directory used for the app
+	Home string `yaml:"home"`
+
+	// KeyringBackend is the default keyring backend to use for blockchain initialization
+	KeyringBackend string `yaml:"keyring-backend"`
+}
+
+// Host keeps configuration related to started servers.
+type Host struct {
+	RPC     string `yaml:"rpc"`
+	P2P     string `yaml:"p2p"`
+	Prof    string `yaml:"prof"`
+	GRPC    string `yaml:"grpc"`
+	GRPCWeb string `yaml:"grpc-web"`
+	API     string `yaml:"api"`
+}
+
+// BaseConfig defines a struct with the fields that are common to all config versions.
+type BaseConfig struct {
+	Version  Version                `yaml:"version"`
+	Build    Build                  `yaml:"build"`
+	Accounts []Account              `yaml:"accounts"`
+	Faucet   Faucet                 `yaml:"faucet,omitempty"`
+	Client   Client                 `yaml:"client,omitempty"`
+	Genesis  map[string]interface{} `yaml:"genesis,omitempty"`
+}
```

Comment:
`BaseConfig` represents the common fields between v0 and v1, but nothing prevents a v2 to have nothing in common with that, and so `BaseConfig` may become no longer usable. 

Same for sub-types, today `Account` type is OK for both v0 and v1, but nothing prevents a newer version to require something different.

With that in mind I would to remove all that types, and no matter if we repeat their definition in v0 and v1.

That makes the migration more verbose on the other hand because you can't reuse types :P

Reply:
It is true that makes it more verbose but I am not sure about this one. What you comment crossed my mind but refactoring it might not be worth it at this point, I think the config format won't change that much over time and if it does we can refactor it at that point when a significant format change happens.